src/web/mjs/connectors/LineMangaJP.mjs

```diff
@@ -51,12 +51,370 @@ export default class LineMangaJP extends Connector {
 
     async _getPages(chapter) {
         const request = new Request(`${this.url}/book/viewer?id=${chapter.id}`, this.requestOptions);
-        const imgs = await Engine.Request.fetchUI(request, 'window.imgs || {}');
-        if (Object.keys(imgs).length == 0) {
+        const [images, option] = await Engine.Request.fetchUI(request, '[Object.values(window.imgs || {}), window.OPTION || {}]');
+        if (option.mediado_token) {
+            const mediadoData = await this._getMediadoData(option);
+            return mediadoData.images.map(imageData => {
+                const metaData = mediadoData.metaDataList[imageData.pageNum - 1];
+                return this.createConnectorURI({
+                    imageData,
+                    metaData,
+                });
+            });
+        }
+        if (images.length == 0) {
             throw new Error(`The chapter '${chapter.title}' is neither public, nor purchased!`);
         }
-        return Object.values(imgs)
-            .map(img => img.url)
+        return images
+            .map(image => image.url)
             .filter(url => !url.includes('inline_ads_banner'));
     }
-}
\ No newline at end of file
+
+    async _handleConnectorURI(payload) {
+        const canvas = document.createElement('canvas');
+        canvas.width = payload.imageData.width;
+        canvas.height = payload.imageData.height;
+        const imageDataUrl = await this._getImageDataUrl(payload.imageData.url);
+        const imageData = {
+            ...payload.imageData,
+            url: imageDataUrl,
+        };
+        await this._unscrambleMediadoImage(imageData, payload.metaData, canvas);
+        const blob = await this._canvasToBlob(canvas);
+        return this._blobToBuffer(blob);
+    }
+
+    async _getImageDataUrl(url) {
+        const uri = new URL(url);
+        const request = new Request(uri);
+        request.headers.set('x-origin', uri.origin);
+        const res = await fetch(request);
+        const blob = await res.blob();
+        return this._blobToDataUrl(blob);
```

Comment:
I think data URLs have a limit of around ~2MB in chrome, so it may not work for larger images.
Never tried, but maybe Object URLs could be used as workaround (but they need to be released at some point to prevent memory leaks).

Reply:
It seems "Image" object has no limit for dataURL. However, there is a ~2MB limit for iframe tags or opening in a new tab.
(tested [here](https://codepen.io/Elastic1/pen/zYWGKrM))